src/components/Panel/PanelContainer.js

```diff
@@ -142,18 +152,50 @@ export default class PanelContainer extends Component {
       secondaryButton,
       subtitle,
       title,
+      hasScrollingContent,
+      panelAriaLabel,
+
+      // Explicitly define so they aren't in `...other`.
+      disableEscape,
+      labels,
+      rootNode,
+
+      // Required because it may include `aria-label`, etc.
+      ...other
     } = this.props;
 
     const hasFooter = renderFooter || primaryButton;
 
+    const ariaLabel =
+      title || this.props['aria-label'] || panelAriaLabel || subtitle;
+
+    const getAriaLabelledBy = title ? this.paneltitleId : this.panelSubtitleId;
+
+    const hasScrollingContentProps = hasScrollingContent
```

Comment:
I'm unsure about this, purely because we may not be able to accurately determine the length of the content being inserted into the panel. Is this to prevent the panel from being focusable if the content is not overflowing?

As a very basic example, if you reduce the height of the viewport, the panel will become scrollable after initialization.

Reply:
There's no way to 100% guarantee, like with the `ScrollGradient` -- you have to trust the consumer to know if they plan to add a lot of content or not. Otherwise, adding attributes that may ultimately be unnecessary, in content, (by just adding all of this attributes by default, for example) would likely be disruptive.

Here's where Carbon added the same logic: https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/Modal/Modal.js#L367

As with the aria-label logic, I think it makes sense to create the same expectations in the `Panel` that Carbon has created for the `Modal`.

Reply:
For additional context, here's an example of the same `Panel` with the same data model, but 2 different results - 

### Without scrolling

<img width="1676" alt="1-2_Panel" src="https://user-images.githubusercontent.com/3846874/72514327-61ecda80-3846-11ea-9ce9-329a4854234c.png">

### With scrolling

<img width="1680" alt="2-2_Panel" src="https://user-images.githubusercontent.com/3846874/72514335-6618f800-3846-11ea-9314-120adc74a494.png">

In this case, I hypothesize that consumers might need to hack together logic to get the height of the content in the panel relative to the rest of it after fetching it, and also account for resizing to determine whether `hasScrollingContent` needs to be applied.

I'm not strongly opposed to this idea, I'm just concerned that this may be a recurring issue for consumers adopting this change and that they might have to implement a solution similar to the above.

I'm wondering if there's a suitable person that works with the panel that we can bounce this off?

Reply:
@SimonFinney  If you set `hasScrollingContent` to `true`, you would have to provide an `aria-label`. Otherwise you have a bunch of other aria props and a tabindex that aren't right without the label. For a consumer, maybe it would be appropriate for them to set `hasScrollingContent` to `true` on their end and ensure they provide a valid label, without even trying to access height of content, which wouldn't be performant.

But I just wanted to also reiterate that this is how things are implemented in the `Modal` currently, and I think it makes sense to keep them consistent. I don't see a reason here to deviate from the upstream library...

Reply:
I don't think we should deviate from the upstream library, but I'm definitely conscious of the effect this may have on existing consumers, dynamic content as highlighted above, and E2E tests.

Perhaps, the best route here is to have `hasScrollingContent` default to `true` as you suggested on Slack!

Reply:
@SimonFinney Yeah did mention that but unfortunately it would be a breaking change 😅 It would throw a proptypes error because if you set `hasScrollContent={true}`, you have to also provide an `aria-label`. So this would cause issues for existing implementations.

Reply:
As I understand it, if there are any (hypothetical) tests expecting to be able to keyboard-tab to the content, this will also affect it, right?

Reply:
Yes when writing a test for this you'd have to ensure that `hasScrollingContent={true}` before running the `userEvent` tab test otherwise it would fail.